lib/util/periods.js

```diff
@@ -734,6 +735,24 @@ shaka.util.PeriodCombiner = class {
           used = false;
         }
       }
+      // In cases where we have the same video codec but different
+      // codec bases in matched representations across periods
+      // (e.g. HEVC as hev and hvc), we don't want to delete the matched stream
```

Comment:
I might be misunderstanding, but it seems like the solution could be just to normalize video codecs in streams before processing periods and in configs before applying preferences.  If they all said hev or all said hvc, problem solved, correct?

Reply:
That could work, but I think would still result in confusion in some cases, like we are finding with manifests returned from Google DAI. 

For example, we have HEVC content encoded in a `hvc` format. When requesting a stitched manifest from DAI, with DAI pre-rolls and our original content, the HEVC pre-roll representations from DAI are encoding in a `hev` format. As a content owner, I know there are `hvc` representations in my manifest but may not know DAI has used a different format for pre-rolls. If Shaka Player "converts" `hvc` to `hev`, then when no `hvc` variants exist it's confusing as to why those aren't available.

Outside the scope of this change but related: In the same manifest we also have content encoded as `dvhe`. DAI does not encode ads in DVHE. Shaka Player matches the content period `dvhe` streams with the pre-roll `hev` streams. `dvhe` variants are still created even though DVHE streams aren't available in pre-rolls. For conformity with that behavior, I think it makes sense to create unique variants for each codec base.

@dsparacio Any additional thoughts here?

Reply:
I'm going on leave, so @theodab and @avelad should make a decision about this without waiting for me.

Reply:
I agree! @theodab what do you think?

Reply:
My thoughts on this.   

My main concern is the difference with HVC and DVHE when mixed with HEV from DAI content.  

1.  As Will pointed out, currently, if we set preferredVideoCodec to [hev, hvc] we get a bitrate ladder with 14 reps instead of 7 and need to filter, but all are marked as HEV even though some are HVC. 

1. When I use preferredVideoCodec[hev, dvhe] I get same as aboven and it excludes DVHE altogether. 

1. When I use preferredVideoCodec[dvhe] I get exactly what I am expecting, 7 bitrates all DVHE representations. 

So as a user, I would expect that using  preferredVideoCodec[hvc] would result the same as preferredVideoCodec[dvhe].  However it only returns 3 items due to the issue we pointed out in the ticket. 

It does seem it would be more efficient to do this the way Joey suggested.  I just want to make sure we have some consistency in the properties settings and expected results.  

Meaning either works for us.... but its odd if they are different IMHO

```
 preferredVideoCodec to [hev, dvhe] - does not return any DVHE (same result as hev, hvc)
 preferredVideoCodec to [hev, hvc] - this returns twice as many bitrates as expected since DAI does not exactly match at resolution.  
```

Or 
 
```
 preferredVideoCodec to [dvhe] works as expected today
 preferredVideoCodec to [hvc] - truncated list returned.
```

Currently api settings

```
 preferredVideoCodec to [dvhe] 
 preferredVideoCodec to [hev, hvc] - have to filter this on our side. 
```

Reply:
@willdharris any update?

Reply:
We were curious to hear @theodab's thoughts on this before moving forward. If the consensus is to go with Joey's suggestion of normalizing codecs earlier, we can start looking at that.

Reply:
@theodab is busy, but I confirm that the entire maintainer team wants this solution, please continue with the development.

Reply:
@willdharris any update? Thanks!